respa_admin/forms.py

```diff
@@ -358,3 +361,61 @@ def _get_images_formset_translated_fields(images_formset, lang_postfix):
             images_translation_count += len([x for x in form.initial if x.endswith(lang_postfix)])
 
     return images_translation_count
+
+
+class UserForm(forms.ModelForm):
+
+    class Meta:
+        model = User
+
+        fields = [
+            'is_staff',
+        ]
+
+        widgets = {
+            'is_staff': RespaGenericCheckboxInput(attrs={
+                'label': _('Staff account'),
+                'help_text': _('Allows user to grant permissions to units')
+            })
+        }
+
+
+class UnitAuthorizationForm(forms.ModelForm):
+    can_approve_reservation = forms.BooleanField(widget=RespaGenericCheckboxInput, required=False)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        can_approve_initial_value = False
+        if self.instance.pk:
+            can_approve_initial_value = self.instance.authorized.has_perm('unit:can_approve_reservation', self.instance.subject)
+        self.fields['can_approve_reservation'].initial = can_approve_initial_value
+
+    class Meta:
+        model = UnitAuthorization
+
+        fields = [
+            'subject',
+            'level',
+            'authorized',
+        ]
+
+
+class UnitAuthorizationFormSet(forms.BaseInlineFormSet):
```

Comment:
Actually, there seems to be performance problems when opening a user with 150 UnitAuthorizations.
Performance could probably be improved by defining a queryset here which includes some `.select_related` or `.prefetch_related` calls to avoid hundreds of queries when the formset is rendered.

Reply:
It seems you can probably override `get_queryset(self)` here, get the queryset that `super().get_queryset()` returns and apply prefetches to that.

https://github.com/django/django/blob/42b9a23267f14be39b9b00958e18d5746783208e/django/forms/models.py#L628